### PR TITLE
[TT-5398] Implement OAS apiKey scheme import

### PR DESCRIPTION
**Pull request by furkansenharputlu**

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4063/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2367993106)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4063/merge
Commit: a60e69ceb2ed0f10b2f7d1176c2e452b28213731 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2367999743)

File: apidef/oas/default.go [hunk 1]
```diff
@@ -7,18 +7,22 @@ import (
...
+
+	errEmptySecurityObject = errors.New("security object is empty in OAS")
 )
 
 type TykExtensionConfigParams struct {
-	UpstreamURL string
-	ListenPath  string
+	UpstreamURL    string
+	ListenPath     string
+	Authentication bool
```
Comment (jeffy-mathew) [inline apidef/oas/default.go:25]: shall we keep `Authentication` flag a ptr `*bool`, thereby we can decide whether the query parameter was not supplied(nil). Both `false` and `true` can be a valid value for `authentication` query param, hence we cannot rely on boolean default value `false`

**Review by lghiur — APPROVED** [review]

**Review by jeffy-mathew — APPROVED** [review]

File: apidef/oas/default.go (hunk 1 above)
Comment (furkansenharputlu) [reply apidef/oas/default.go]: Added

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4063/merge
Commit: 745840b96536a0168f81a6e9564d44a611aa935b 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2383939140)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 63849cfdbbb55252c23150fbfbac4b6aee746717 [TT-5398] Implement OAS apiKey scheme import (#4063)

* Implement OAS apiKey scheme import

* Make Authentication bool a pointer
Triggered by: push (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2384021273)

**Pull request merged**